### ISSUE TITLE
Please don't forget dvfs operating points

### ISSUE THREAD

**Issue by ThomasKaiser** [longsleep/build-pine64-image#4]
Regarding #3 you should keep in mind that in the dvfs table currently the defined maximum is set to 1152MHz @ 1300mV. Without touching this I wouldn't unlock the cooler table to clock up to 1344.

BTW: OMFG. They shipped with the same insane [1536GHz](https://github.com/longsleep/linux-pine64/blob/lichee-dev-v3.10.65-fix-compile/drivers/cpufreq/sunxi-cpufreq.c#L48) as in the H3 BSP. I bet it will take a week until the first moron adjusts VDD_CPUX to 1.5 to be able to repeat the unfortunate H3 story: A64 blamed to be a crap SoC due to insane overvolting/overclocking ;)

**Comment by longsleep** [comment]
Yeah i saw that - thanks for pointing it out. I take a look into the cooling / clock / voltage situation and probably add some easy to use monitoring scripts probably taking some bits from the RPi monitoring config you posted.

**Comment by ThomasKaiser** [comment]
After playing around with this reliability stuff for approx. 1.5 years now and given we're talking about an Allwinner SoC with BSP kernel implementing this budget cooling (behaves identical to A83T/H3) I would strongly recommend to install RPi-Monitor if you start to adjust the cooler table.

It's kinda fascinating how thermal behaviour differs if you adjust settings. And I would suspect there's a bug in the settings used now since I've never seen clockspeed decreases below 816MHz or killed CPU cores in situations where this would've been happened on H3 way before.

**Comment by longsleep** [comment]
I briefly looked int RPi-Monitor and found it cumbersome to install/configure. So i hacked a bit and created a script (https://github.com/longsleep/build-pine64-image/blob/master/simpleimage/pine64_health.sh)

```
sudo ./pine64_health.sh 
CPU freq       : 480.00 GHz 
CPU count      : 4  
Govenor        : ondemand  
Core voltage   : 1.04 V 
SOC Temp       : 46 C 
Cooling state  : 0  
Cooling limit  : roomage:0,0,0,0,1344000,4,0,0 
```

Not as nice with graphs and history, but no installation and helps with quick testing.

**Comment by ThomasKaiser** [comment]
The script is nice and I already stole and adopted it for H3, A20 and A83T ;)

But RPi-Monitor's adavantage are the RRD databases. You start a test, go to dinner, the board deadlocks and after the reboot you'll have a look into the graphs ;)

Anyway: In case you start with a Debian based image from scratch I would be interested if http://kaiser-edv.de/tmp/4U4tkD/install-rpi-monitor-for-a64.sh throws errors or installs everything flawlessly...

**Comment by longsleep** [comment]
@ThomasKaiser i tried your install-rpi-monitor script - it installed fine but the web interface fails with a Javascript error: Uncaught TypeError: Cannot read property 'length' of undefined in rpimonitor.js:295.

**Comment by ThomasKaiser** [comment]
@longsleep thx, I started from scratch with your Xenial image and can reproduce it now. Will have a look into it and try to come up with a solution.

**Comment by ThomasKaiser** [comment]
Now it works, I created the template package wrong and therefore RPi-Monitor got stuck (the client component running in the browser obviously not able to deal with the garbage fed from the server task). Simply installing again should work:

![bildschirmfoto 2016-03-02 um 12 14 58](https://cloud.githubusercontent.com/assets/12658961/13458930/858431be-e070-11e5-8246-075e48cdf31c.png)

And checking this box is important of course ;)

![bildschirmfoto 2016-03-02 um 12 30 54](https://cloud.githubusercontent.com/assets/12658961/13459261/ba91cd88-e072-11e5-8514-a1c42ec649b0.png)

**Comment by longsleep** [comment]
Cool, i will not be able to try before tomorrow though :/

**Comment by longsleep** [comment]
@ThomasKaiser - RPi-Monitor works fine now (just ran the script again).

**Comment by longsleep** [comment]
Fixed with 1ab7a47e5d19375fd47f6748ad73e509bea2c251
